src/pudl/workspace/datastore.py

```diff
@@ -1,715 +1,408 @@
-    domain = domains.pop()
-    ftp = ftplib.FTP(domain)  # nosec: B321 Sadly required for ferc1 & epacems
-    login_result = ftp.login()
-    assert login_result.startswith("230"), \
-        f"Failed to login to {domain}: {login_result}"
-    url_to_retry = []
-    tmp_to_retry = []
-    error_messages = []
-    for path, tmp_file, src_url in zip(within_domain_paths, tmp_files, src_urls):
-        with open(tmp_file, "wb") as f:
-            try:
-                ftp.retrbinary(f"RETR {path}", f.write)
-            except ftplib.all_errors as e:
-                error_messages.append(e)
-                url_to_retry.append(src_url)
-                tmp_to_retry.append(tmp_file)
-    # Now retry failures recursively
-    num_failed = len(url_to_retry)
-    if num_failed > 0:
-        if allow_retry and len(src_urls) == 1:
-            # If there was only one URL and it failed, retry once.
-            return _download_ftp(url_to_retry, tmp_to_retry, allow_retry=False)
-        elif allow_retry and src_urls != url_to_retry:
-            # If there were multiple URLs and at least one didn't fail,
-            # keep retrying until all fail or all succeed.
-            return _download_ftp(url_to_retry,
-                                 tmp_to_retry,
-                                 allow_retry=allow_retry)
-        if url_to_retry == src_urls:
-            err_msg = (
-                f"Download failed for all {num_failed} URLs. " +
-                "Maybe the server is down?\n" +
-                "Here are the failure messages:\n " +
-                " \n".join(error_messages)
-            )
-        if not allow_retry:
-            err_msg = (
-                f"Download failed for {num_failed} URLs and no more " +
-                "retries are allowed.\n" +
-                "Here are the failure messages:\n " +
-                " \n".join(error_messages)
-            )
-        warnings.warn(err_msg)
-
-
-def _download_default(src_urls, tmp_files, allow_retry=True):
-    """Download URLs to files. Designed to be called by `download` function.
-
-    If the file cannot be downloaded, the program will issue a warning.
-
-    Args:
-        src_urls (list of str): the source URLs to download.
-        tmp_files (list of str): the corresponding files to save.
-        allow_retry (bool): Should the function call itself again to
-            retry the download? (Default will try twice for a single file, or
-            until all files fail)
-    Returns:
-        None
-
-    Todo:
-        Replace assert statement
-
-    """
-    assert len(src_urls) == len(tmp_files) > 0
-    url_to_retry = []
-    tmp_to_retry = []
-    for src_url, tmp_file in zip(src_urls, tmp_files):
+            self._dois = DOI["production"]
+            self.token = TOKEN["production"]
+            self.api_root = "https://zenodo.org/api"
+
+        with PUDL_YML.open() as f:
+            cfg = yaml.load(f.read(), Loader=yaml.FullLoader)
```

Comment:
Codacy found an issue: [Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5759597)